#! /bin/sh -e
## 11_handle_propfind.dpatch by Tollef Fog Heen <tfheen@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Don't fall flat on our face if we get a request type which we
## DP: don't understand.

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch ${2:+-d $2}}"

if [ $# -lt 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $patch_opts -p1 < $0;;
    -unpatch) patch $patch_opts -p1 -R < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0
@DPATCH@

--- mailman-2.1.4.orig/scripts/driver
+++ mailman-2.1.4/scripts/driver
@@ -81,6 +82,15 @@
         module = getattr(pkg, scriptname)
         main = getattr(module, 'main')
         try:
+            import os
+            request_method = os.environ.get('REQUEST_METHOD')
+            if not request_method in ['GET', 'POST', 'HEAD']:
+                print "Status: 405 Method not allowed"
+                print "Content-type: text/plain"
+                print
+                print "The method is not allowed"
+                sys.exit()
+                
             try:
                 sys.stderr = logger
                 sys.stdout = tempstdout
